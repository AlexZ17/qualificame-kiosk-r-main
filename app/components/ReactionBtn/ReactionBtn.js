import React, { PureComponent } from 'react';
import { 
	Text,
	View,
	TouchableOpacity,
	Image
} from 'react-native';
import PropTypes from 'prop-types';

import styles from './ReactionBtnStyle';

export default class ReactionBtn extends PureComponent {
	render() {
		let imgSrc;
		switch (this.props.type) {
			case 'excellent': 
				imgSrc = require('qualificame-kiosk/assets/excellent.png');
				break;
			case 'average':
				imgSrc = require('qualificame-kiosk/assets/average.png');
				break;
			case 'bad':
				imgSrc = require('qualificame-kiosk/assets/bad.png');
				break;
			case 'awful':
				imgSrc = require('qualificame-kiosk/assets/awful.png');
				break;
			default:
				imgSrc = require('qualificame-kiosk/assets/excellent.png');
				break;
		}
		let size = {
			width: this.props.size,
			height: this.props.size
		}
		return (
			<TouchableOpacity activeOpacity={0.7} onPress={() => this.props.onPress()}>
				<Image source={imgSrc} style={size}/>
			</TouchableOpacity>
		);
	}
}

	ReactionBtn.propTypes = {
		size: PropTypes.number,
		onPress: PropTypes.func
	}

	ReactionBtn.defaultProps = {
		size: 100,
		onPress:  () => {}
	}
