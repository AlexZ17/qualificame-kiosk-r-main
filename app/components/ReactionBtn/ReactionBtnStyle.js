import { StyleSheet } from 'react-native';
import { ApplicationStyles, Fonts, Colors, Metrics } from 'qualificame-kiosk/app/styles';

export default StyleSheet.create({
});
