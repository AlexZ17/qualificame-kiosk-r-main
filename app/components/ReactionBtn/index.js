import ReactionBtn from './ReactionBtn';

export default ReactionBtn;
