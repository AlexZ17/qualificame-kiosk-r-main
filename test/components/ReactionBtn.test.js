import React from 'react';
import { shallow } from 'enzyme';
import {
	TouchableOpacity,
	Image
} from 'react-native';

import ReactionBtn from 'qualificame-kiosk/app/components/ReactionBtn';
import image from 'qualificame-kiosk/assets/excellent.png'

function createTestProps (props) {
	return {
		onPress: ()=>{},
		...props,
	}
}

describe('rendering', () => {
	let wrapper = shallow(<ReactionBtn />);
	it('renders as expected', () => {
		expect(wrapper).toMatchSnapshot();
	});
	it('renders size the image', () => {
		expect(wrapper.find(Image).props().style.width ).toEqual(100);
		expect(wrapper.find(Image).props().style.height ).toEqual(100);
	});
	it('renders image', () => {
		expect(wrapper.find(Image).prop('source')).toEqual(image)
	});
});

describe('interactions', () => {
	let wrapper;
	const createWrapper = props => shallow(<ReactionBtn {...props}/>);
	describe('when onPress function is passed to props', () => {
		let fn;
		beforeEach(() => {
			fn = jest.fn();
			const props = createTestProps({ onPress: fn })
			wrapper = createWrapper(props);
		});
		it('it is called on button press', () => {
			wrapper.find(TouchableOpacity).first().props().onPress();
			expect(fn).toBeCalled();
		});
	});
});