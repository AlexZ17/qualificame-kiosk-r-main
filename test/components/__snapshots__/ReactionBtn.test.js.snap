// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering renders as expected 1`] = `
<TouchableOpacity
  activeOpacity={0.7}
  onPress={[Function]}
>
  <Component
    source={1}
    style={
      Object {
        "height": 100,
        "width": 100,
      }
    }
  />
</TouchableOpacity>
`;
